XenAdmin/Actions/GUIActions/DownloadUpdatesXmlAction.cs

```diff
@@ -313,5 +322,26 @@ protected virtual XmlDocument FetchCheckForUpdatesXml(string location)
 
             return xdoc;
         }
+
+        private string GetUniqueIdHash()
+        {
+            string uniqueIdHash = "nil";
+
+            var managementObj = new System.Management.ManagementObject("Win32_OperatingSystem=@");

```

Comment:
I suppose all these system calls are OK, but I can't be sure that none of them can throw exceptions. Maybe we should wrap the whole of lines 330-342 in a try-catch block.